Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt

```diff
@@ -0,0 +1,80 @@
+package de.rki.coronawarnapp.dccticketing.core.common
+
+import com.nimbusds.jose.JWSAlgorithm.ES256
+import com.nimbusds.jose.JWSAlgorithm.PS256
+import com.nimbusds.jose.JWSAlgorithm.RS256
+import com.nimbusds.jose.crypto.ECDSAVerifier
+import com.nimbusds.jose.crypto.RSASSAVerifier
+import com.nimbusds.jose.crypto.bc.BouncyCastleProviderSingleton
+import com.nimbusds.jose.util.X509CertUtils
+import com.nimbusds.jwt.SignedJWT
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException.ErrorCode.JWT_VER_ALG_NOT_SUPPORTED
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException.ErrorCode.JWT_VER_NO_JWKS
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException.ErrorCode.JWT_VER_NO_JWK_FOR_KID
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException.ErrorCode.JWT_VER_NO_KID
+import de.rki.coronawarnapp.dccticketing.core.common.DccTicketingException.ErrorCode.JWT_VER_SIG_INVALID
+import de.rki.coronawarnapp.dccticketing.core.transaction.DccJWK
+import okio.ByteString.Companion.decodeBase64
+import org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPublicKey
+import timber.log.Timber
+import java.security.PublicKey
+import java.security.interfaces.RSAPublicKey
+import javax.inject.Inject
+
+class DccJWKVerification @Inject constructor() {
+
+    init {
+        X509CertUtils.setProvider(BouncyCastleProviderSingleton.getInstance())
+    }
+
+    fun verify(jwt: String, jwkSet: Set<DccJWK>) {
+        // 1. Check for empty jwkSet
+        if (jwkSet.isEmpty()) throw DccTicketingException(JWT_VER_NO_JWKS)
```

Comment:
should we create another exception like `DccTicketingServerCertificateCheckException` so we can map it to 
`DccTicketingException` in other places ?

With this we can just change the error code ,which I think can lead to issues , I prefer if create a totally different exception with these error codes of verification what do you think ?

Reply:
`DccTicketingJwtException` has been created